internal/config/specs.go

```diff
@@ -23,6 +23,14 @@ type EnvSpec struct {
 	HydraAdminURL       string `envconfig:"hydra_admin_url" required:"true"`
 	OathkeeperPublicURL string `envconfig:"oathkeeper_public_url" required:"true"`
 
+	AuthenticationEnabled           bool     `envconfig:"authentication_enabled" default:"false" validate:"required"`
+	OIDCIssuer                      string   `envconfig:"oidc_issuer" validate:"required"`
+	OAuth2ClientId                  string   `envconfig:"oauth2_client_id" validate:"required"`
+	OAuth2ClientSecret              string   `envconfig:"oauth2_client_secret" validate:"required"`
+	OAuth2RedirectURI               string   `envconfig:"oauth2_redirect_uri" validate:"required"`
+	OAuth2CodeGrantScopes           []string `envconfig:"oauth2_codegrant_scopes" default:"openid,offline_access" validate:"required"`
+	AccessTokenVerificationStrategy string   `envconfig:"access_token_verification_strategy" default:"jwks" validate:"oneof=jwks userinfo"`
```

Comment:
Should we have a common prefix for the oauth2 vars?

Reply:
I tried to keep it consistent, thing is, the only one missing the oauth2 prefix is the verification strategy (which already has a long name) so I left it like that since "access token" is already explicitly oauth2 related.
OIDC issuer I prefer to keep it OIDC prefixed for its meaning, and the first AuthenticationEnabled I didn't put the prefix there because authn is at a higher conceptual level that OAuth2.
So basically I could add the prefix to the verification strategy making it `OAuth2AccessTokenVerificationStrategy`, I'm open to it